### PR TITLE
first setyp sync documentation

### PR DESCRIPTION
**Pull request by Sieuwkje-firely**
I still need to add examples for messages to the broker. Ewout has offered to help out with these. Please check if I am not mentioning anything silly, or if parts are unclear.

### CODE REVIEW

**Comment by alexzautke** [comment]
@cknaap Would you mind doing the review here?

File: features_and_tools/pubsub.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+* ``SendLightEvents`` - Either set to true or false, this will enable Firely Server to send out messages with information on changes, these messages will not contain the complete resource.
+* ``SendFullEvents`` - Either set to true or false, this will enable Firely Server to send out messages with information on changes, these messages contain the complete resource.
+* ``PollingIntervalSeconds`` - By default set to five seconds, Firely Server will check the database every five seconds for changes. 
+* ``MaxPublishBatchSize`` - The maximum amount of resources that can be send to the message broker at once.
+
+
+Message types and formats
+-------------------------
+
+PubSub expects standard message formats for a set of different messages that can be communicated to RabbitMQ. When other services want to receive updates on changes they can expect messages also to take on a set format.
```
Comment (eugene-blokhin) [inline features_and_tools/pubsub.rst:74]: I was thinking about rephrasing it to something like...
```suggestion
To establish communication between Firely Server and other applications, the parties must share the same contract. This section describes the contract.
```

File: features_and_tools/pubsub.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+
+Message types and formats
+-------------------------
+
+PubSub expects standard message formats for a set of different messages that can be communicated to RabbitMQ. When other services want to receive updates on changes they can expect messages also to take on a set format. 
+PubSub uses MassTransit to envelope these messages using specific headers, for documentation on these headers we refer to the `MassTransit documentation page <https://masstransit.io/documentation/concepts/messages#message-headers>`_.
+Below we give an overview of the different message types that can be communicated with PubSub.
+
+ExecuteStorePlan
```
Comment (eugene-blokhin) [inline features_and_tools/pubsub.rst:78]:
```suggestion
ExecuteStorePlanCommand
```

File: features_and_tools/pubsub.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+^^^^^^^^^^^^^^^^
+
+This message can be send to the message broker by your client to let Firely Server execute a batch of instructions to create, update, upsert, or delete resources that should be processed as a transaction, so either
```
Comment (eugene-blokhin) [inline features_and_tools/pubsub.rst:81]:
```suggestion
This command can be sent to the message broker by your client to let Firely Server execute a batch of instructions to create, update, upsert, or delete resources that should be processed as a transaction, so either
```

File: features_and_tools/pubsub.rst [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+            "resource": "{\"resourceType\":\"Patient\",\"id\":\"testid\",\"meta\":{\"versionId\":\"versionId=test\",\"lastUpdated\":\"2023-10-09T12:00:22.8990506+02:00\"},\"name\":[{\"family\":\"id=test\"}]}",
+            "resourceType": "",
+            "resourceId": "",
+            "currentVersion": "",
+            "operation": create
+        }
+        ]
+    }
+
+The ``ExecuteStorePlan`` message typically contains an array of instructions, where each instruction can contain the following fields:
```
Comment (eugene-blokhin) [inline features_and_tools/pubsub.rst:101]:
```suggestion
The ``ExecuteStorePlanCommand`` message typically contains an array of instructions, where each instruction can contain the following fields:
```

File: features_and_tools/pubsub.rst [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+      }
+    ]
+  },
+
+  "expirationTime": "2023-10-15T10:36:55.0495563Z",
+  "sentTime": "2023-10-15T10:35:54.8989292Z",
+  "headers": {
+    "MT-Request-AcceptType": [
+      "urn:message:Firely.Server.Contracts.Messages.V1:ExecuteStorePlanResponse"
+    ]
```
Comment (i1zolotariov) [inline features_and_tools/pubsub.rst:122]:
```suggestion
    ],
```

**Review by Sieuwkje-firely — COMMENTED** [review]
I think it is such a big improvement compared to what I had, I really like the "Click to expand" sections, they make it very organized. Your adjustments and additions make it much clearer

I think this may be part of another ticket for documentation, but it may be good to indicate the names of the exchanges that are automatically generated by Firely Server upon first startup, maybe together with a screenshot from how it looks in RabbitMQ. 
Maybe this should be part of a RabbitMQ tutorial, but it may also be good to note that in order to get the responses from FS for the RetrievePlanCommand and ExecuteStorePlanCommand, separate exchanges need to be created in RabbitMQ (that can be used in the ResponseAddress header of the earlier command messages) that need to be bound to their own queue if the user wants to get these messages. Similarly, for the ResourcesChangedLightEvent and ResourcesChangedEvent, the respective exchanges that are generated by FS need to be bound to a Queue if a user wants to retrieve these messages.

I am talking from my own perspective here, the one with no experience with RabbitMQ, so for the users that will implement this it may already be clear that this creation of exchanges and binding is necessary.

**Pull request merged**